src/screens/ChatScreen/index.js

```diff
@@ -27,59 +27,149 @@ import {
+        if (data.nation_id >= 0) {
+          this.setState({joined: true});
+          this.connection.on('msg', (messageData) => {
+            const messages = [{
+              _id: messageData._id,
+              text: messageData.msg,
+              createdAt: messageData.createdAt,
+              user: {
+                _id: messageData.userId,
+                name: messageData.from
+              }
+            }];
+            this.setState(previousState => ({ messages: GiftedChat.append(previousState.messages, messages) }));
+          });
+        }
+      });
+    } else {
+      // add initial bot message
+      this.setState({
+        messages: [
+          {
+            _id: 1,
+            text: elizabot.start(),
+            createdAt: new Date(),
+            user: {
+              _id: 2,
+              name: 'Dr. FreudBot',
+            },
+            // Any additional custom parameters are passed through
+          },
+        ],
+      });
+    }
+  }
 
-  componentWillMount() {
-    this.setState({
-      messages: [
+  onSend(messages = []) {
+    if (this.props.isBot) {
+      const m = [
         {
-          _id: 1,
-          text: elizabot.start(),
+          _id: this.state.messages.length + 1,
+          text: elizabot.reply(messages[0].text),
           createdAt: new Date(),
           user: {
             _id: 2,
-            name: 'Eliza',
+            name: 'Dr. FreudBot',
           },
-          // Any additional custom parameters are passed through
-        },
-      ],
-    });
-  }
-
-  onSend(messages = []) {
-    const m = [
-      {
-        _id: this.state.messages.length + 1,
-        text: elizabot.reply(messages[0].text),
-        createdAt: new Date(),
-        user: {
-          _id: 2,
-          name: 'Eliza',
         },
-      },
-    ];
-
-    // Add user's message
-    this.setState(previousState => ({ messages: GiftedChat.append(previousState.messages, messages) }));
-
-    // Add Eliza's response
-    this.setState(previousState => ({ messages: GiftedChat.append(previousState.messages, m) }));
+      ];
+
+      // Add user's message
+      this.setState(previousState => ({ messages: GiftedChat.append(previousState.messages, messages) }));
+
+      // Add Eliza's response
+      this.setState(previousState => ({ messages: GiftedChat.append(previousState.messages, m) }));
+    } else {
+      if (this.state.joined) {
+        const newMessage = {
+          nation_id: this.props.nationId,
+          msg: messages[0].text,
+          from: this.props.user ? this.props.user.name : 'anonymous',
+          userId: this.props.user ? this.props.user.uid : 'anonymous',
+        }
+        this.connection.emit('room:msg', newMessage);
+      } else {
+
+      }
+    }
   }
 
   render() {
+    const sendingUser = {
+      _id: this.props.user ? this.props.user.uid : 'anonymous',
+      name: this.props.user ? this.props.user.name : 'anonymous',
```

Comment:
Duplicated user generation logic, please extract.

Reply:
Actually I added this logic in order to handle the exception. Basically  a user should not be able to enter chat screen without creating a user profile.

Reply:
Okay then.